server/aws-lsp-codewhisperer/src/language-server/agenticChat/context/contextCommandsProvider.ts

```diff
@@ -24,33 +18,13 @@ export class ContextCommandsProvider implements Disposable {
         private readonly lsp: Lsp
     ) {
         this.registerPromptFileWatcher()
-        this.lsp.workspace.onDidCreateFiles(async event => {
-            const filePaths = event.files.map(file => URI.parse(file.uri).fsPath)
-            await this.updateContextCommandsOnFileChange(filePaths, true)
-        })
-        this.lsp.workspace.onDidDeleteFiles(async event => {
-            const filePaths = event.files.map(file => URI.parse(file.uri).fsPath)
-            await this.updateContextCommandsOnFileChange(filePaths, false)
-        })
-        this.lsp.workspace.onDidRenameFiles(async event => {
-            const oldFiles = event.files.map(file => URI.parse(file.oldUri).fsPath)
-            await this.updateContextCommandsOnFileChange(oldFiles, false)
-            const newFiles = event.files.map(file => URI.parse(file.newUri).fsPath)
-            await this.updateContextCommandsOnFileChange(newFiles, true)
-        })
-    }
 
-    async updateContextCommandsOnFileChange(filePaths: string[], isAdd: boolean) {
-        try {
-            const localProjectContextController = await LocalProjectContextController.getInstance()
-            const shouldUpdate = await localProjectContextController.shouldUpdateContextCommand(filePaths, isAdd)
-            if (shouldUpdate) {
-                const contextItems = await localProjectContextController.getContextCommandItems()
+        void (async () => {
```

Comment:
can you keep the try catch so that there won't be uncaught exceptions from promoises

Reply:
plus it makes sense to await here.. why void